raw_linux.go

```diff
@@ -297,6 +297,29 @@ func (p *packetConn) SetPromiscuous(b bool) error {
 	return p.s.SetSockopt(unix.SOL_PACKET, membership, unsafe.Pointer(&mreq), unix.SizeofPacketMreq)
 }
 
+// SetMulticast joins or drops membership of data link layer multicast address on interface,
+// allowing it to recieve multicast traffic.
+func (p *packetConn) SetMulticast(b bool, addr net.HardwareAddr) error {
+	// Convert hardware address back to byte array form
+	var baddr [8]byte
```

Comment:
On Linux, it's probably time to refactor out a helper that has a signature like so:

```go
func rawHardwareAddr(addr net.HardwareAddr) ([8]byte, error)
```

Then the helper can do the length check, copy the bytes into the array, and return the array.